### PR TITLE
El 193 autocode triggers

### PR DESCRIPTION
**Pull request by ykalra**
Added generator functions for Triggers.h and Triggers.cpp

### CODE REVIEW

File: autocoding/generators.py [hunk 1]
```diff
@@ -1,71 +1,140 @@
...
+        out += (
+            "State* state"
+            + trigger.attrib["currState"]
+            + "To"
+            + trigger.attrib["nextState"]
+            + "Trigger() "
+        )
+        for cond in trigger:
+            out += (
+                "{ return ("
```
Comment (bennyE31) [inline autocoding/generators.py]: I think we should consider the case where there are more than 2 conditions that trigger a true return

**Review by bennyE31 — CHANGES_REQUESTED** [review]
This looks really good so far! There are some more things I would like to see, though.

I think there will be cases in a real implementation of the state machine where there are more than 2 conditions that need to be checked. I left a more specific comment on the relevant line of code.

We should also include a template file for these files, they will probably less complicated than the existing data_h template. It should only need the necessary include statements.

This is a larger, more abstract change, but I was thinking that since the Triggers.cpp and Triggers.h files are so similar. could they be built from the same XML? This would simplify what we would need to write when we add new triggers. What are your thoughts on something like this?

File: autocoding/generators.py (hunk 1 above)
Comment (ykalra) [reply autocoding/generators.py]: I created the template files while testing. I could push them so you could look over them. Triggers.cpp and Triggers.h are based on the same xml file. I couldn't figure out how to use the new xml file with autocode so I moved the xml code for Triggers to data.xml and I'll look over the comments on the code by tomorrow.

**Comment by bennyE31** [comment]
Sadly, this request is going to be closed. Due to some changes in electrical hardware that we couldn't foresee at the beginning of the semester, we will no longer be using a state machine.

**Pull request closed without merging**